contracts/interfaces/blade/IBridgeGateway.sol

```diff
@@ -0,0 +1,57 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.19;
+
+/**
+ * @param _address address of the validator
+ * @param blsKey BLS public key
+ * @param votingPower voting power of the validator
+ */
+struct Validator {
+    address _address;
+    uint256[4] blsKey;
+    uint256 votingPower;
+}
+
+/**
+ * @param id id of message
+ * @param sourceChainId id of source chain
+ * @param destinationChainId id of destination chain
+ * @param sender sender account of this bridge message
+ * @param receiver receiver account of this bridge message
+ * @param payload payload
+ */
+struct BridgeMessage {
+    uint256 id;
+    uint256 sourceChainId;
+    uint256 destinationChainId;
+    address sender;
+    address receiver;
+    bytes payload;
+}
+
+/**
+ * @param messages list of all messages in batch
+ * @param sourceChainId id of chain which is source of batch
+ * @param destinationChainId id of chain which is destination of batch
+ */
+struct BridgeMessageBatch {
+    BridgeMessage[] messages;
+    uint256 sourceChainId;
```

Comment:
Are we going to have mixed source chain ids in a single batch? If that's the case, then I'll remove this field or at least make it an array.

On the other hand, I believe that we've agreed that a single `BridgeMessageBatch` contains only bridge messages for the same destination chain, meaning that `destinationChainId` would be the same (so we can leave the `destinationChainId`, even though it is somewhat redundant, because it can be retrieved from the arbitrary `BridgeMessage`).

Reply:
No, only single source and single destination chain messages in a batch. I would leave it so we can have a sanity check. If this is same as in all messages, then the batch is valid. Plus, we won't have to access message in the batch to get the source or destination chain id.